[quant] Per channel quantization support in insert_prepack_unpack

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29494 [quant][graphmode] Returning axis from calculate_qparams
* #29493 [quant][graphmode] Test for per channel graph mode quantization
* #29416 [quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`
* **#29701 [quant] Per channel quantization support in insert_prepack_unpack**
* #29492 [quant][graphmode] Support per channel quantization in insert_quant_dequant and fold_prepack

Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580908](https://our.internmc.facebook.com/intern/diff/D18580908)

jerryzh168 committed:
[quant] Per channel quantization support in insert_prepack_unpack

Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Per channel quantization support in insert_prepack_unpack"


Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on test/test_jit.py:
```diff
@@ -1176,9 +1196,22 @@ def forward(self, x):
-        torch._C._jit_pass_insert_prepack_unpack(graph)
-        FileCheck().run(conv_input_str, graph)
+
+        conv_per_channel = """
```
This test is useless, it just repeats the pattern from the code. If the pattern happens to be buggy, this test will still pass. Can we rewrite it in a similar way to the test above? If not, it's better to just remove it.

jerryzh168 replied:
I think even the above code is pretty similar to just repeating the pattern, maybe only way to test this properly is through  integration test.

jerryzh168 replied:
should we just remove this test then?

ZolotukhinM replied:
By the code above I meant the code where you create actual modules - these tests are fine. But here we're literally creating the same IR as in the pattern - it's fine to delete this if there is no way to rewrite it like the code with modules.

jerryzh168 replied:
reason we do this is in Linear we are matching CallFunction and in Conv2d we are matching aten::conv2d. I think we can write this test in similar ways as the Linear test.

ZolotukhinM reviewed (approved):
The change looks good, but please revisit the tests before landing.

jerryzh168 committed:
Update on "[quant] Per channel quantization support in insert_prepack_unpack"


Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Per channel quantization support in insert_prepack_unpack"


Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Per channel quantization support in insert_prepack_unpack"


Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Per channel quantization support in insert_prepack_unpack"


Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Per channel quantization support in insert_prepack_unpack"


Summary:
att

Test Plan:
python test/test_jit.py 'TestJit.test_insert_prepack_unpack'

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580908](https://our.internmc.facebook.com/intern/diff/D18580908)

[ghstack-poisoned]
